draft-ietf-cbor-7049bis.md

```diff
@@ -408,27 +409,31 @@ floats as integers or vice versa, perhaps to save encoded bytes.
 
 A CBOR data item  ({{cbor-data-models}}) is encoded to or decoded from
 a byte string as described in this section.  The encoding is
-summarized in {{jumptable}}.
+summarized in {{jumptable}}.  An encoder MUST produce only well-formed
```

Comment:
That is a weird MUST.  Maybe we should instead define CBOR encoder as one that produces well-formed CBOR?

The practical issue is with CBOR encoders that take a CBOR-encoded data item, enclose it (not as a string, but as a pre-encoded subtree) into a larger CBOR-encoded data item, and ship that.  I don't think we should mandate that such CBOR encoders always have to check all enclosed pre-encoded data items.

Reply:
I think the effect of saying "An encoder MUST" is to contribute to a definition of "CBOR encoder" as the programs that happen to do this.

The practical issue is also an issue with your new text in #56: You wrote, `An encoder MUST NOT issue two-byte sequences that start with 0xf8`, but an encoder that blindly encloses a pre-encoded subtree wouldn't ensure that. I think we should be able to MUST encoders, with maybe the result that partial encoders that blindly copy subtrees are only "CBOR encoders" when you consider them in combination with the programs that produce those subtrees.